src/levanter/data/text.py

```diff
@@ -788,10 +796,23 @@ def build_caches(
             if weight == 0 and split == "train":
                 continue
 
-            source_config_dict = source_config.__dict__
+            source_config_dict = dict(**source_config.__dict__)
```

Comment:
Add comment on why we need to delete it - to make the LMDatasetMixtureComponentConfig act like a LMDatasetSourceConfig?

Reply:
more or less. i also cleaned this part up a bit